ons_alpha/private_media/utils.py

```diff
@@ -0,0 +1,131 @@
+    name: str
+
+
+class PrivateRootCollectionNotFound(ObjectDoesNotExist):
+    pass
+
+
+class PublicRootCollectionNotFound(ObjectDoesNotExist):
+    pass
+
+
+def get_root_collection_info() -> dict[str, CollectionDetails]:
+    """
+    Returns a basic representation of the 'Private' and 'Public' root collections created by a data migration in the
+    core app.
+
+    Raises `PrivateRootCollectionNotFound` if the 'Private' collections has not yet been created (or has been deleted).
+
+    Raises `PublicRootCollectionNotFound` if the 'Public' collection has not yet be created (or has been deleted).
+    """
+    try:
+        return _ROOT_COLLECTION_INFO.value
+    except AttributeError:
+        pass
+
+    info = {}
+    for obj in Collection.objects.filter(depth=2).values_list("id", "path", "name", named=True):
+        if "private" not in info and obj.name.lower() == "private":
+            info["private"] = obj
+        if "public" not in info and obj.name.lower() == "public":
+            info["public"] = obj
+
+    if "private" not in info:
+        raise PrivateRootCollectionNotFound(
+            "The 'Private' Collection cannot be found below 'Root' as expected. Did you forget to run migrations?"
+        )
+    if "public" not in info:
+        raise PublicRootCollectionNotFound(
+            "The 'Public' Collection cannot be found below 'Root' as expected. Did you forget to run migrations?"
+        )
+
+    # Cache result for the current thread / asyncio task
+    _ROOT_COLLECTION_INFO.value = info
+    return info
+
+
+def get_private_root_collection_id() -> int:
+    return get_root_collection_info()["private"].id
+
+
+def get_public_root_collection_id() -> int:
+    return get_root_collection_info()["public"].id
+
+
+def get_private_root_collection_path() -> str:
+    return get_root_collection_info()["private"].path
+
+
+def get_public_root_collection_path() -> str:
+    return get_root_collection_info()["public"].path
+
+
+def collection_is_private(collection: Collection) -> bool:
+    return collection.path.startswith(get_private_root_collection_path())
+
+
+def collection_is_public(collection: Collection) -> bool:
+    return not collection_is_private(collection)
+
+
+def set_file_acl(file: FieldFile, acl_name: str) -> bool:
+    """
+    Set the ACL for the supplied `file` to the supplied `acl_name` value.
+    Any exceptions raised during the processed are captured and logged.
+    Returns `True` if no errors occured, otherwise `False`.
+    """
+    if not isinstance(file.storage, S3Storage):
+        # In environments that don't use S3, return True immediately,
+        # as no futher action is needed, and there's nothing to go wrong!
+        logger.info(f"ACL setting is unnecessary for {file.name}")
+        return True
+
+    obj = file.storage.bucket.Object(file.name)
+    try:
+        obj_acl = obj.Acl()
+    except ClientError as e:
+        logger.debug(f"Failed to retreive ACL for {file.name}: {e!r}")
```

Comment:
Issue: Use `logger.exception` to properly capture the exception, rather than trying to stringify it in the message. This also correctly reports to tools like Sentry.

Reply:
Yes, that is how it was implemented originally... However, I'm concerned about raising exceptions for things that the code simply cannot avoid... Especially when we already have a plan for retrying. I'm not sure it's something that would be useful in the longer term (I think it would just be an annoyance).